src/csharp/Microsoft.Spark/Sql/SparkSession.cs

```diff
@@ -101,6 +101,46 @@ public void Dispose()
         public DataFrameReader Read() =>
             new DataFrameReader((JvmObjectReference)_jvmObject.Invoke("read"));
 
+        /// <summary>
+        /// Creates a DataFrame with a single column named id, containing elements in
+        /// a range from 0 to end (exclusive) with step value 1.
+        /// as a DataFrame.
+        /// </summary>
+        /// <returns>DataFrame object</returns>
+        public DataFrame Range(long end) =>
+            new DataFrame((JvmObjectReference)_jvmObject.Invoke("range", end));
+
+        /// <summary>
+        /// Creates a DataFrame with a single column named id, containing elements in 
+        /// a range from start to end (exclusive) with step value 1.
+        /// </summary>
+        /// <returns>DataFrame object</returns>
+        public DataFrame Range(long start, long end) =>
+            new DataFrame(
+                (JvmObjectReference)_jvmObject.Invoke("range", start, end));
+
+        /// <summary>
+        /// Creates a DataFrame with a single column named id, containing elements in
+        /// a range from start to end (exclusive) with a step value.
+        /// </summary>
+        /// <returns>DataFrame object</returns>
+        public DataFrame Range(long start, long end, long step) =>
+            new DataFrame(
+                (JvmObjectReference)_jvmObject.Invoke(
+                    "range", start, end, step));
+
+        /// <summary>
+        /// Creates a DataFrame with a single column named id, containing elements in
+        /// a range from start to end (exclusive) with a step value, with partition
+        /// number specified.
+        /// </summary>
+        /// <returns>DataFrame object</returns>
+        public DataFrame Range(long start, long end, long step, int numPartitions) =>
```

Comment:
On this one, the signatures are:

long end
long start, long end
long start, long end, long step
long start, long end, long step, numPartitions

If we just have one - should I move end before start so that it doesn't have a default or default end as well to 0?

so either:
long start = 0, long end = 0, long? step = null, int? numPartitions = null
long end, long start = 0, long? step = null, int? numPartitions = null

I am not sure switching end and start is right, maybe defaulting start and end as well?

Reply:
cool sorry didn't see the last comment, so leave the different signatures?

Reply:
yeah you can leave as is.